src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -800,5 +809,17 @@ private bool InferNamedPipesInformation()
 
         private static bool IsLocalHost(string serverName)
             => ".".Equals(serverName) || "(local)".Equals(serverName) || "localhost".Equals(serverName);
+
+        private string GetInstanceNameFromDataSource()
+        {
+            string instanceName = string.Empty;
+            string[] tokensByBackSlash = _dataSourceAfterTrimmingProtocol.Split(BackSlashCharacter);
+            if (tokensByBackSlash.Length > 1)
+            {
+                instanceName = tokensByBackSlash[1];
```

Comment:
There are also other places in the class, where tokens are not trimmed, can you fix those too?

Reply:
Found all usage of token then use Trim() function when used.